src/prefect/engine/flow_runner.py

```diff
@@ -155,8 +155,26 @@ def initialize_run(  # type: ignore
         context.update(flow_name=self.flow.name)
         context.setdefault("scheduled_start_time", pendulum.now("utc"))
 
+        # Determine the current time, allowing our formatted dates in the context
+        # to rely on a manually set value
+        now = context.get("date")
+        if isinstance(now, str):
+            # Attempt to parse into a `DateTime` object
+            try:
+                now = pendulum.parse(now)
```

Comment:
When parsing the date, should we also convert to `'utc'` so the `date` context is always in utc? No strong thoughts either way.

Reply:
I think not. It wouldn't override their `date` variable since we're using `setdefault` below and it'd make `date` be in whatever tz they used and all the dependent variables in `utc`

Reply:
>  It wouldn't override their date variable since we're using setdefault below

Hmmm, I'm wondering if we should override `date` in context if we successfully parse it. Values set via cloud are sent as json (so it's a string), but if `date` is normally a pendulum object it'd be nice to keep that interface uniform.

Reply:
I think it makes sense to override it in just the case where we parse from string -> pendulum object, I'll add that.